streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -406,6 +407,23 @@ final class ZSink[-R, +E, -In, +L, +Z] private (val channel: ZChannel[R, ZNothin
     ZSink.unwrapScoped(scoped)
   }
 
+  def refineOrDie[E1](
+    pf: PartialFunction[E, E1]
+  )(implicit ev1: E <:< Throwable, ev2: CanFail[E], trace: Trace): ZSink[R, E1, In, L, Z] =
```

Comment:
We should use `isSubtypeOfError` here for improved error messages.

Reply:
Updated here and also for ZStream